src/Query/Traits/JoinComponentCompiler.php

```diff
@@ -11,39 +11,48 @@ trait JoinComponentCompiler
     /**
      * Compiles join to string to pass this string in query.
      *
-     * @param Builder    $query
-     * @param JoinClause $join
+     * @param Builder      $query
+     * @param JoinClause[] $joins
+     *
+     * @throws GrammarException
      *
      * @return string
      */
-    protected function compileJoinComponent(Builder $query, JoinClause $join): string
+    protected function compileJoinsComponent(Builder $query, array $joins): string
     {
-        $this->verifyJoin($join);
-
         $result = [];
 
-        if ($join->isDistributed()) {
-            $result[] = 'GLOBAL';
-        }
+        foreach ($joins as $join) {
+            $this->verifyJoin($join);
 
-        if (!is_null($join->getStrict())) {
-            $result[] = $join->getStrict();
-        }
+            if ($join->isDistributed()) {
+                $result[] = 'GLOBAL';
+            }
 
-        if (!is_null($join->getType())) {
-            $result[] = $join->getType();
-        }
+            if (!is_null($join->getStrict())) {
+                $result[] = $join->getStrict();
+            }
+
+            if (!is_null($join->getType())) {
```

Comment:
Accordingly to the new join syntax the type should be before strictness.

Reply:
But just now i realised that this change could lead to a significant tests rewrite:( You'll need to rewrite all joins tests with specified strictness.
So, if the latest clickhouse accepts this old order of join parts - forget about it:)

Reply:
Yeah, latest versions of ClickHouse supports the old order of join parts.